doc/releases/changelog-dev.md

```diff
@@ -4,6 +4,49 @@
 
 <h3>New features since last release</h3>
 
+* `qml.ops.Sum` now supports storing grouping information. Grouping type and method can be
+  specified during construction using the `grouping_type` and `method` keyword arguments of
+  `qml.dot`, `qml.sum`, or `qml.ops.Sum`. The grouping indices are stored in `Sum.grouping_indices`.
+  [(#5179)](https://github.com/PennyLaneAI/pennylane/pull/5179)
+
+  ```python
+  import pennylane as qml
+
+  a = qml.PauliX(0)
+  b = qml.PauliX(1)
+  c = qml.PauliZ(0)
+  obs = [a, b, c]
+  coeffs = [1.0, 2.0, 3.0]
+
+  op = qml.dot(coeffs, obs, grouping_type="qwc")
+  ```
+  ```pycon
+  >>> op.grouping_indices
+  ((0, 1), (2,))
+  ```
```

Comment:
Perhaps this could be made more clear that this points to the ops in `terms()` and not `ops` explicitly.
Would also be good to document these examples either in `sum`, `Sum` itself, or `Sum.compute_grouping`